app/models/metric/ci_mixin.rb

```diff
@@ -20,12 +20,7 @@ module Metric::CiMixin
       virtual_column vcol, :type => :float, :uses => :vim_performance_operating_ranges
     end
 
-    supports :capture do
-      metrics_capture_klass = "#{self.class.module_parent.name}::MetricsCapture".safe_constantize
-      unless metrics_capture_klass&.method_defined?(:perf_collect_metrics)
-        _('This provider does not support metrics collection')
-      end
-    end
+    supports_not :capture
```

Comment:
Going to switch this to `supports_not` at the base classes like we do for other features, and `supports :capture` only in classes that support it.

These are the models which `include Metric::CiMixin`
```
app/models/miq_region.rb
app/models/service.rb
app/models/zone.rb
app/models/storage.rb
app/models/ext_management_system.rb
app/models/host.rb
app/models/vm_or_template.rb
app/models/availability_zone.rb
app/models/container.rb
app/models/container_group.rb
app/models/container_image.rb
app/models/container_node.rb
app/models/container_project.rb
app/models/container_replicator.rb
app/models/container_service.rb
app/models/ems_cluster.rb
app/models/host_aggregate.rb
app/models/miq_enterprise.rb
```

And this is the full list of classes we need to add `supports :capture` to
```
 ci_mixin_classes.flat_map do |klass|
?>   klass.descendants.select { |k| klass_name = "#{k.module_parent}::MetricsCapture"; klass_name.safe_constantize&.name == klass_name }
>> end.map(&:name)
=> 
["ManageIQ::Providers::Vmware::InfraManager::OvfService",
 "ManageIQ::Providers::Vmware::InfraManager::Storage",
 "ManageIQ::Providers::Ovirt::InfraManager::Storage",
 "ManageIQ::Providers::Ovirt::InfraManager::IsoDatastore",
 "ManageIQ::Providers::Redhat::InfraManager::Storage",
 "ManageIQ::Providers::Redhat::InfraManager::IsoDatastore",
 "ManageIQ::Providers::IbmPowerHmc::InfraManager::Storage",
 "ManageIQ::Providers::IbmPowerHmc::InfraManager::MediaRepository",
 "ManageIQ::Providers::Vmware::InfraManager::Host",
 "ManageIQ::Providers::Vmware::InfraManager::HostEsx",
 "ManageIQ::Providers::IbmPowerHmc::InfraManager::Host",
 "ManageIQ::Providers::Ovirt::InfraManager::Host",
 "ManageIQ::Providers::Redhat::InfraManager::Host",
 "ManageIQ::Providers::Openstack::InfraManager::Host",
 "ManageIQ::Providers::Vmware::InfraManager::Vm",
 "ManageIQ::Providers::IbmPowerHmc::InfraManager::Vm",
 "ManageIQ::Providers::IbmPowerHmc::InfraManager::Lpar",
 "ManageIQ::Providers::IbmPowerHmc::InfraManager::Vios",
 "ManageIQ::Providers::Ovirt::InfraManager::Vm",
 "ManageIQ::Providers::Redhat::InfraManager::Vm",
 "ManageIQ::Providers::Amazon::CloudManager::Vm",
 "ManageIQ::Providers::Azure::CloudManager::Vm",
 "ManageIQ::Providers::AzureStack::CloudManager::Vm",
 "ManageIQ::Providers::Google::CloudManager::Vm",
 "ManageIQ::Providers::IbmCloud::VPC::CloudManager::Vm",
 "ManageIQ::Providers::Openstack::CloudManager::Vm",
 "ManageIQ::Providers::IbmCic::CloudManager::Vm",
 "ManageIQ::Providers::IbmPowerVc::CloudManager::Vm",
 "ManageIQ::Providers::OracleCloud::CloudManager::Vm",
 "ManageIQ::Providers::Vmware::InfraManager::Template",
 "ManageIQ::Providers::IbmPowerHmc::InfraManager::Template",
 "ManageIQ::Providers::Ovirt::InfraManager::Template",
 "ManageIQ::Providers::Redhat::InfraManager::Template",
 "ManageIQ::Providers::Openstack::InfraManager::Template",
 "ManageIQ::Providers::Amazon::CloudManager::Template",
 "ManageIQ::Providers::Azure::CloudManager::Template",
 "ManageIQ::Providers::Google::CloudManager::Template",
 "ManageIQ::Providers::IbmCloud::VPC::CloudManager::Template",
 "ManageIQ::Providers::Openstack::CloudManager::BaseTemplate",
 "ManageIQ::Providers::Openstack::CloudManager::Template",
 "ManageIQ::Providers::IbmCic::CloudManager::Template",
 "ManageIQ::Providers::IbmPowerVc::CloudManager::Template",
 "ManageIQ::Providers::Openstack::CloudManager::VolumeSnapshotTemplate",
 "ManageIQ::Providers::Openstack::CloudManager::VolumeTemplate",
 "ManageIQ::Providers::OracleCloud::CloudManager::Template",
 "ManageIQ::Providers::Amazon::CloudManager::AvailabilityZone",
 "ManageIQ::Providers::Azure::CloudManager::AvailabilityZone",
 "ManageIQ::Providers::AzureStack::CloudManager::AvailabilityZone",
 "ManageIQ::Providers::Google::CloudManager::AvailabilityZone",
 "ManageIQ::Providers::IbmCloud::VPC::CloudManager::AvailabilityZone",
 "ManageIQ::Providers::Openstack::CloudManager::AvailabilityZone",
 "ManageIQ::Providers::Openstack::CloudManager::AvailabilityZoneNull",
 "ManageIQ::Providers::IbmCic::CloudManager::AvailabilityZone",
 "ManageIQ::Providers::IbmCic::CloudManager::AvailabilityZoneNull",
 "ManageIQ::Providers::IbmPowerVc::CloudManager::AvailabilityZone",
 "ManageIQ::Providers::IbmPowerVc::CloudManager::AvailabilityZoneNull",
 "ManageIQ::Providers::OracleCloud::CloudManager::AvailabilityZone",
 "ManageIQ::Providers::Kubernetes::ContainerManager::Container",
 "ManageIQ::Providers::IbmCloud::ContainerManager::Container",
 "ManageIQ::Providers::Kubernetes::ContainerManager::ContainerGroup",
 "ManageIQ::Providers::IbmCloud::ContainerManager::ContainerGroup",
 "ManageIQ::Providers::Kubernetes::ContainerManager::ContainerNode",
 "ManageIQ::Providers::IbmCloud::ContainerManager::ContainerNode",
 "ManageIQ::Providers::Vmware::InfraManager::Cluster",
 "ManageIQ::Providers::Ovirt::InfraManager::Cluster",
 "ManageIQ::Providers::Redhat::InfraManager::Cluster",
 "ManageIQ::Providers::Openstack::InfraManager::Cluster",
 "ManageIQ::Providers::Openstack::CloudManager::HostAggregate",
 "ManageIQ::Providers::IbmCic::CloudManager::HostAggregate",
 "ManageIQ::Providers::IbmPowerVc::CloudManager::HostAggregate"]
```

Reply:
Total aside, but capture is kind of a terrible name for metrics collection  We should consider a different name, like `:metrics_capture` or even just `:metrics`

Reply:
Yeah I like just `supports :metrics`, we already have simply `supports :events`